models/articles.models.js

```diff
@@ -1,14 +1,21 @@
-const { rows } = require("pg/lib/defaults");
+const res = require("express/lib/response");
 const db = require("../db/connection");
 
 const selectArticleById = (id) => {
   return db
-    .query(`SELECT * FROM articles WHERE article_id = $1`, [id])
+    .query(
+      `SELECT articles.*, COUNT(comments.article_id = $1) AS comment_count FROM articles LEFT JOIN comments ON comments.article_id = articles.article_id GROUP BY  articles.article_id`,
+      [id]
+    )
     .then((result) => {
```

Comment:
Have a look at CAST in SQL on W3 schools for how you can implement a conversion of your comment_count to INT